### PR TITLE
[PSM interop] Change restart policy to Never

### PR DESCRIPTION
**Pull request by lidizheng**
In an offline discussion, we discovered that the auto-restart of crashed pods is hiding potential issues. We want to turn off the auto-restart of pods, so setting the `restartPolicy: Never` to client and server pods.

* [grpc_xds_url_map](http://sponge/3f29b42b-5229-40df-8769-ee3561f6a1d3)
* [grpc_xds_url_map_python](http://sponge/11756861-7931-4578-8dd3-823cdc62a32e)
* [grpc_xds_k8s_lb](http://sponge/5d9e2fda-6caf-4af5-b558-fe4ee184f51b)
* [grpc_xds_k8s_lb_python](http://sponge/d5dad7f3-d051-4e3a-8dcd-cadccc6b04e3)
* [grpc_xds_k8s_xlang](http://sponge/aaeb82b3-638d-44d9-b4f1-fccae05a5e14)
* [psm-security](http://sponge/bba550b0-d8b7-4c06-97aa-eef8a229bd73)
* [psm-security-python](http://sponge/0b102d4c-b473-4bfe-8a72-bee89a6fbad7)

### CODE REVIEW

**Review by yashykt — APPROVED** [review]

**Comment by lidizheng** [comment]
Interestingly, the pod-level restart policy only support `Always`, despite what the Kubernetes document says: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy

> kubernetes.utils.create_from_yaml.FailToCreateError: Error from server (Unprocessable Entity): {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Deployment.apps \"psm-grpc-client\" is invalid: spec.template.spec.restartPolicy: Unsupported value: \"Never\": supported values: \"Always\"","reason":"Invalid","details":{"name":"psm-grpc-client","group":"apps","kind":"Deployment","causes":[{"reason":"FieldValueNotSupported","message":"Unsupported value: \"Never\": supported values: \"Always\"","field":"spec.template.spec.restartPolicy"}]},"code":422}

This is an existing issue, see https://github.com/kubernetes/kubernetes/issues/24725. So, I changed the policy scope to containers (the client/server container).

File: tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/server.deployment.yaml [hunk 1]
```diff
@@ -21,10 +21,12 @@ spec:
       % if service_account_name:
       serviceAccountName: ${service_account_name}
       % endif
+      restartPolicy: Never
       containers:
       - name: ${deployment_name}
         image: ${image_name}
         imagePullPolicy: Always
+        restartPolicy: Never
```
Comment (sergiitk) [inline tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/server.deployment.yaml:29]: @lidizheng did you mean to remove this too?

Comment (lidizheng) [reply tools/run_tests/xds_k8s_test_driver/kubernetes-manifests/server.deployment.yaml:28]: Good catch. Removed the pod-level `restartPolicy` field.

**Review by sergiitk — APPROVED** [review]

**Comment by lidizheng** [comment]
Ad-hoc tests are passing. Merging.

**Pull request merged**